OlinkAnalyze/R/bridgeable.R

```diff
@@ -0,0 +1,293 @@
+    message(paste0("Finding assay correlation in 3K and HT"))
+
+    data_out <- data |>
+      dplyr::group_by(.data[["Gene"]]) |>
+      dplyr::reframe(outlier_removal_iqr(dplyr::pick(everything()),
+                                         iqr_value = 3)) |>
+      dplyr::group_by(.data[["Gene"]]) |>
+      filter(.data[["Count_3k"]] > 10, .data[["Count_ht"]] > 10) |>
+      dplyr::mutate(R2_lm = stats::cor(x = .data[["NPX_3k"]],
+                                       y = .data[["NPX_ht"]])^2) |>
+      dplyr::ungroup() |>
+      dplyr::distinct()
+
+    return(data_out)
+  }
+
+
+  ks_results <- function(data = data) {
+    message(paste0("Finding normalization method for bridgeable assays"))
+
+    data_out <- data |>
+      dplyr::group_by(.data[["Gene"]], .data[["ids"]]) |>
+      dplyr::reframe(outlier_removal_iqr(dplyr::pick(everything()),
+                                         iqr_value = 3)) |>
+      dplyr::group_by(.data[["Gene"]], .data[["ids"]]) |>
+      filter(.data[["Count_3k"]] > 10, .data[["Count_ht"]] > 10) |>
+      dplyr::summarise(ks = stats::ks.test(
+        .data[["NPX_ht"]], .data[["NPX_3k"]])$statistic, .groups = "keep") |>
+      dplyr::ungroup() |>
+      dplyr::distinct()
+
+    return(data_out)
+  }
+
+
+  e3k_eHT_mapping <- OlinkAnalyze:::eHT_e3072_mapping
+
+  data_Explore384 <- data_Explore384 |>
+    dplyr::filter(
+      stringr::str_detect(string = .data[["SampleType"]], pattern = "SAMPLE"),
+      stringr::str_detect(string = .data[["AssayType"]], pattern = "assay")
+    ) |>
+    map_oid_ht_3k() |>
+    dplyr::mutate(OlinkID_Explore384 = substr(.data[["OlinkID"]], 10, 18)) |>
+    dplyr::rename(
+      Gene = .data[["Assay"]],
+      NPX_3k = .data[["NPX"]],
+      ids = .data[["OlinkID"]]
+    ) |>
+    dplyr::select(.data[["SampleID"]],
+                  .data[["ids"]],
+                  .data[["OlinkID_Explore384"]],
+                  .data[["NPX_3k"]],
+                  .data[["Count"]]) |>
+    dplyr::rename(Count_3k = .data[["Count"]]) |>
+    distinct()
+
+  # Count refers to HT counts
+  data_HT <- data_HT |>
+    dplyr::filter(
+      stringr::str_detect(string = .data[["SampleType"]], pattern = "SAMPLE"),
+      stringr::str_detect(string = .data[["AssayType"]], pattern = "assay")
+    ) |>
+    map_oid_ht_3k() |>
+    dplyr::rename(
+      Gene = .data[["Assay"]],
+      NPX_ht = .data[["NPX"]],
+      ids = .data[["OlinkID"]]
+    ) |>
+    dplyr::mutate(OlinkID = substr(.data[["ids"]], 1, 8)) |>
+    dplyr::select(.data[["SampleID"]],
+                  .data[["Gene"]],
+                  .data[["OlinkID"]],
+                  .data[["ids"]],
+                  .data[["UniProt"]],
+                  .data[["NPX_ht"]],
+                  .data[["Count"]]) |>
+    dplyr::mutate(Count_ht = .data[["Count"]]) |>
+    distinct()
+
+  data_mapped <- data_Explore384 |>
+    dplyr::inner_join(data_HT, by = c("SampleID", "ids"),
+                      relationship = "many-to-many") |>
```

Comment:
Chunks of code 147-165 and 168-188 prepare the two datasets "data_Explore384" and "data_HT". These two chunks repeat each other and we can skip them by merging the input datasets and doing tidyr::pivot_wider. I think this is a more elegant approach which takes full advantage of the tidyverse stack.

Reply:
These chunks have been re-done with a pivot_wider and combined with the data_mapped chunk